File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int res = 0;
        int tmp = x;
        int sgn = ((tmp < 0) ? -1 : 1);
        if(tmp == INT_MIN) return 0;
        tmp = abs(tmp);
        while(tmp) {
            if(res > INT_MAX / 10 || (res == INT_MAX / 10 && tmp % 10 >= 7)) return 0;
            res = res * 10 + (tmp % 10);
            tmp = tmp / 10;
        }
        return sgn * res;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        imax = (1 << 31) - 1
        imin = -(1 << 31)
        inp = str(x)[1:] if x < 0 else str(x)
        sgn = -1 if x < 0 else 1
        res = int(inp[::-1]) * sgn
        if res < imin or res > imax: 
            return 0
        return res
        